### PR TITLE
Eliminate deprecation warnings

### LINKED ISSUE #19: Eliminate warnings from test run
Some of these may be outside our control, in which case we can suppress them; others we should fix:
```
../env/lib/python3.10/site-packages/_pytest/fixtures.py:227
  /home/alec/r/pytket-qiskit/env/lib/python3.10/site-packages/_pytest/fixtures.py:227: PendingDeprecationWarning: The qiskit.Aer entry point will be deprecated in a future release and subsequently removed. Instead you should use this directly from the root of the qiskit-aer package.
    fixturemarker: Optional[FixtureFunctionMarker] = getattr(

tests/backend_test.py::test_operator
tests/backend_test.py::test_operator
tests/backend_test.py::test_operator
tests/backend_test.py::test_operator
tests/backend_test.py::test_expectation_bug
tests/backend_test.py::test_aer_placed_expectation
tests/backend_test.py::test_aer_placed_expectation
  /home/alec/r/pytket-qiskit/pytket/extensions/qiskit/backends/aer.py:710: PendingDeprecationWarning: The PauliTable class has been superseded by PauliList and is pending deprecation. This class will be deprecated in the future release and subsequently removed after that.
    return SparsePauliOp(PauliTable(table_array), coeffs)

tests/backend_test.py::test_operator
tests/backend_test.py::test_expectation_bug
tests/backend_test.py::test_aer_placed_expectation
tests/backend_test.py::test_aer_placed_expectation
  /home/alec/r/pytket-qiskit/env/lib/python3.10/site-packages/qiskit/quantum_info/operators/symplectic/sparse_pauli_op.py:131: ComplexWarning: Casting complex values to real discards the imaginary part
    self._coeffs = np.asarray((-1j) ** (phase - count_y) * coeffs, dtype=coeffs.dtype)

tests/qiskit_backend_test.py: 12 warnings
  /home/alec/r/pytket-qiskit/env/lib/python3.10/site-packages/qiskit_aer/noise/device/models.py:365: UserWarning: Device model returned an invalid T_2 relaxation time greater than the theoretical maximum value 2 * T_1 (94238.37625242842 > 2 * 2953.117923413739). Truncating to maximum value.
    warn("Device model returned an invalid T_2 relaxation time greater than"

tests/qiskit_backend_test.py::test_qiskit_counts
tests/qiskit_backend_test.py::test_qiskit_counts
  /home/alec/r/pytket-qiskit/env/lib/python3.10/site-packages/qiskit_aer/noise/device/models.py:365: UserWarning: Device model returned an invalid T_2 relaxation time greater than the theoretical maximum value 2 * T_1 (9.423837625242843e-05 > 2 * 2.9531179234137392e-06). Truncating to maximum value.
    warn("Device model returned an invalid T_2 relaxation time greater than"

tests/qiskit_backend_test.py::test_grover
  /home/alec/r/pytket-qiskit/tests/qiskit_backend_test.py:166: PendingDeprecationWarning: The quantum_instance argument has been superseded by the sampler argument. This argument will be deprecated in a future release and subsequently removed after that.
    grover = Grover(quantum_instance=qinstance)

tests/qiskit_backend_test.py::test_unsupported_gateset
  /home/alec/r/pytket-qiskit/tests/qiskit_backend_test.py:184: PendingDeprecationWarning: The quantum_instance argument has been superseded by the sampler argument. This argument will be deprecated in a future release and subsequently removed after that.
    grover = Grover(quantum_instance=qinstance)

tests/qiskit_backend_test.py::test_unsupported_gateset
  /home/alec/r/pytket-qiskit/tests/qiskit_backend_test.py:197: PendingDeprecationWarning: The quantum_instance argument has been superseded by the sampler argument. This argument will be deprecated in a future release and subsequently removed after that.
    grover = Grover(quantum_instance=qinstance)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

### PR DESCRIPTION
**Pull request by cqc-alec**
Closes #19 .

There remains one warning (out of 229 initially), but this is coming from inside qiskit.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -45,7 +45,8 @@
     include_package_data=True,
     install_requires=[
         "pytket ~= 1.21",
-        "qiskit ~= 0.44.0",
+        "qiskit ~= 0.44.3",
```
Comment (cqc-melf) [inline setup.py:48]: Do you want to put this into the changelog?

File: setup.py [hunk 2]
```diff
@@ -45,7 +45,8 @@
...
+        "qiskit-algorithms ~= 0.2.1",
```
Comment (cqc-melf) [inline setup.py:49]: See above

File: setup.py (hunk 1 above)
Comment (cqc-alec) [reply setup.py:48]: Yes, I forgot to commit it...

Comment (cqc-alec) [reply setup.py:48]: Done.

File: setup.py (hunk 2 above)
Comment (cqc-alec) [reply setup.py:49]: Done.

**Review by cqc-melf — DISMISSED** [review]

**Review by cqc-melf — APPROVED** [review]

**Pull request merged**